source/FreeRTOS_IPv6.c

```diff
@@ -686,7 +690,7 @@ eFrameProcessingResult_t eHandleIPv6ExtensionHeaders( NetworkBufferDescriptor_t
          * appear immediately after an IPv6 header only. Outlined
          * by RFC 2460 section 4.1  Extension Header Order.
          */
-        if( ( xExtHeaderCount > 1 ) && ( xCurrentOrder == 1 ) ) /* ipIPv6_EXT_HEADER_HOP_BY_HOP */
```

Comment:
We should check xCurrentOrder here because we might break the loop if ucNextHeader is TCP/UDP/ICMP in line 668.
So I change to check xNextOrder and it must not be hop by hop because xNextOrder implies it's not the first extension header.